sky/jobs/utils.py

```diff
@@ -105,6 +109,14 @@ def get_job_status(backend: 'backends.CloudVmRayBackend',
     return status
 
 
+def _controller_process_alive(pid: int) -> bool:
+    """Check if the controller process is alive."""
+    try:
+        return psutil.Process(pid).is_running()
+    except psutil.NoSuchProcess:
+        return False
```

Comment:
Should we use something similar to our normal job process identification, to avoid the case where a pid is reused? https://github.com/skypilot-org/skypilot/blob/fd1ac0e6fbde830ac6be9e42f523a0e460fd84e8/sky/skylet/job_lib.py#L566-L572